src/arithmetic/arithmetic_expression.c

```diff
@@ -172,10 +172,17 @@ AR_ExpNode *AR_EXP_NewParameterOperandNode(const char *param_name) {
  * e.g. MINUS(X) where X is a constant number will be reduced to
  * a single node with the value -X
  * PLUS(MINUS(A), B) will be reduced to a single constant: B-A. */
-bool AR_EXP_ReduceToScalar(AR_ExpNode *root) {
+bool AR_EXP_ReduceToScalar(AR_ExpNode *root, bool runtime, SIValue *val) {
```

Comment:
I think `val` should be set to NULL on entry and set to scalar only of the entire expression was reduced,
I know this is a recursive function, but you might want to introduce a  `static bool _ AR_EXP_ReduceToScalar` function to help out.

Reply:
setting to NullVal